src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from "react"
+import { postNewComment } from "../utils/utils"
+import Comments from "./Comments"
+
+
+const CommentAdder = ({review_id}) => {
+
+    const [newComment, setNewComment] = useState( {
+        username:'',
+        body: ''
+        })
+
+    const [isPosted, setIsPosted]=useState(true)
+    const [err, setErr] = useState(null)  
+
+
+    function handleCommentChange(event){
+        setNewComment((currComment) => {return {...currComment, [event.target.id]:event.target.value}})
+    }
+
+ 
+    function handleSubmit(event){
+        event.preventDefault()
+        setErr(null)
+   
+        setIsPosted(false)
+       
+        postNewComment(newComment, review_id).then((comment) => {
+            
+            setIsPosted(true);
+            setNewComment((currComment) => {return {username:'', body:''}})
+          }).catch(err => {
+            setIsPosted('not')
+            if(newComment.body.length==0 || newComment.username.length==0){setErr('Comment and username field required')}
+            else{
+            setErr('Something went wrong, please try again.');}
+          });
+         
+            
+    }
```

Comment:
This is not optimistically rendered and waits for a response from your API.
Think about how we can change this function and the what you're showing your comments so that we can display the comment as soon as the button is pressed and then remove it if something goes wrong.

Look at the notes on optimistic rendering.